File: flows/flow-1/scriptlets/+scriptlet#3.ts
//#region generated meta
type Inputs = {
    url: string;
    save_dir: string;
};
type Outputs = {
    file_path: string;
};
//#endregion

import type { Context } from "@oomol/types/oocana";
import fs from "fs";
import path from "path";
import fetch from "node-fetch";

export default async function(
    params: Inputs,
    context: Context<Inputs, Outputs>
): Promise<Partial<Outputs> | undefined | void> {
    const { url, save_dir } = params;
    const response = await fetch(url);
    const buffer = await response.buffer();
    const filename = path.basename(url);
    const save_path = path.join(save_dir, filename);
    fs.writeFileSync(save_path, buffer);
    return { save_path };
};

File: subflows/comic-crawler-download/scriptlets/+scriptlet#3.py
from oocana import Context
import os

#region generated meta
import typing
class Inputs(typing.TypedDict):
    name: str
    save_dir: str
class Outputs(typing.TypedDict):
    pack_path: str | None
#endregion

def main(params: Inputs, context: Context) -> Outputs:
    name = params.get("name")
    save_dir = params.get("save_dir")
    pack_path = os.path.join(save_dir, name) if name and save_dir else None

    return { "pack_path": pack_path }